addons/sale/models/payment_transaction.py

```diff
@@ -146,7 +146,8 @@ def _send_invoice(self):
                 lambda i: not i.is_move_sent and i.state == 'posted' and i._is_ready_to_be_sent()
             )
             invoice_to_send.is_move_sent = True # Mark invoice as sent
-            invoice_to_send.with_user(SUPERUSER_ID)._generate_pdf_and_send_invoice(template)
+            user = self.env.user._is_internal() and self.env.user.id or SUPERUSER_ID
+            invoice_to_send.with_user(user)._generate_pdf_and_send_invoice(template)
```

Comment:
Isn't the issue more with portal than internal users ?
Also, this will impact the user sending the invoice, and it shouldn't be the customer.

Maybe the best solution would be to remove the current user partner from the invoice followers, and make sure the invoicing partner is follower no ?

Reply:
No issue was related to only internal user because user is creator of invoice and partner of invoice (17.0: Case 1)
It wasn't impacting that sender of invoice stays `odoobot` I guess :thinking: 

But where we can do that because `website_sale` does not have any code related to this flow and we can't be sure about removing current user's main contact from follower of invoice will not cause any issue like preventing user from opening invoice record :thought_balloon: 

I was thinking maybe we could create invoice by `odoobot` instead current user it'll not add current user in follower of invoice (But this will also impact normal flows also :slightly_frowning_face: )
Is it okay? :thinking:

Reply:
> In the specification of the ticket, it's said as a known customer, it means a portal user for me, not an internal one, are you sure about that ?

Yes it was also impacting portal user :cry: 

So had to remove current user from follower.
Only remove portal or public user follower so they internal user don't loss access to invoice and check partner of invoice is child of main contact

Also I had an idea to keep follower of current user but uptick `Discussions` subtype but problem is currently there is no way we can do that calling `message_subscribe` with `subtypes=[]` as an agrument always add default subtypes and there is not other method to change subtypes of follower. Is there any other subtype we can tick on invoice instead `Discussions` for case :thinking: 
Also can we add a functionality in Mail to allow adding follower without any subtype :thought_balloon: (This can be useful for other application also where record visibility depends on follower and they don't want to spam users )

Reply:
Mhm, seems quite magical and error-prone to be honest.
I'll confirm it with paju once she's back, but we may want to keep the current behavior, even if it's not perfect.